src/flb_engine.c

```diff
@@ -762,6 +795,19 @@ int flb_engine_start(struct flb_config *config)
                 /* Event type registered by the Scheduler */
                 flb_sched_event_handler(config, event);
             }
+            else if (event->type == FLB_ENGINE_EV_THREAD_ENGINE) {
+                struct flb_output_coro *output_coro;
+
+                /* Read the coroutine reference */
+                ret = flb_pipe_r(event->fd, &output_coro, sizeof(struct flb_output_coro *));
+                if (ret <= 0 || output_coro == 0) {
+                    flb_errno();
+                    continue;
+                }
+
+                /* Init coroutine */
+                flb_coro_resume(output_coro->coro);
```

Comment:
This is all very similar to how workers start coroutines, except that a coroutine is dispatched rather than a worker task. Reason: simplicity, task worker routes/round robin not needed for simple engine dispatch.